src/blockbroadcast.jl

```diff
@@ -0,0 +1,56 @@
+
+# Here we override broadcasting for banded matrices.
+# The design is to to exploit the broadcast machinery so that
+# banded matrices that conform to the banded matrix interface but are not
+# <: AbstractBandedMatrix can get access to fast copyto!, lmul!, rmul!, axpy!, etc.
+# using broadcast variants (B .= A, B .= 2.0 .* A, etc.)
+
+
+abstract type AbstractBlockStyle{N} <: AbstractArrayStyle{N} end
+struct BlockStyle{N} <: AbstractBlockStyle{N} end
+struct PseudoBlockStyle{N} <: AbstractBlockStyle{N} end
+
+
+BlockStyle(::Val{N}) where {N} = BlockStyle{N}()
+PseudoBlockStyle(::Val{N}) where {N} = PseudoBlockStyle{N}()
+BlockStyle{M}(::Val{N}) where {N,M} = BlockStyle{N}()
+PseudoBlockStyle{M}(::Val{N}) where {N,M} = PseudoBlockStyle{N}()
+BroadcastStyle(::Type{<:BlockArray{<:Any,N}}) where N = BlockStyle{N}()
+BroadcastStyle(::Type{<:PseudoBlockArray{<:Any,N}}) where N = PseudoBlockStyle{N}()
+BroadcastStyle(::DefaultArrayStyle{N}, b::AbstractBlockStyle{M}) where {M,N} = typeof(b)(_max(Val(M),Val(N)))
```

Comment:
Are you planning to "vendor" `_max` at some point?  Otherwise it doesn't run on Julia master.